### PR TITLE
Add YodaCondition analyzer

### PR DESCRIPTION
**Pull request by alhimik45**
Hey!

Type:  new feature

Link to issue: #227 

This pull request affects the following components:

* [ ] Core
* [x] Analyzer
* [ ] Compiler
* [ ] Control Flow Graph
* [ ] Documentation

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/ovr/phpsa/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change: This PR adds an analyzer that detects usage of "yoda conditions".

Thanks

### CODE REVIEW

**Comment by ddmler** [comment]
Thanks @alhimik45 :+1:

File: src/Analyzer/Pass/Statement/YodaCondition.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * @param $stmt
+     * @param Context $context
+     * @return bool
+     */
+    public function pass($stmt, Context $context)
+    {
+        $condition = $stmt->cond;
+
+        if ($stmt instanceof For_ && count($stmt->cond) > 0) { // For is the only one that has an array as condition
```
Comment (K-Phoen) [inline src/Analyzer/Pass/Statement/YodaCondition.php]: So if there are several conditions, only the first one is analyzed? Is this wanted?

Comment (ddmler) [inline src/Analyzer/Pass/Statement/YodaCondition.php]: @K-Phoen I would think if anything we should show a message saying that you should make those conditions into one (like you have to do with all other statements that use conditions). It makes no sense to me to have "multiple conditions" if you can merge them with boolean and.

Comment (K-Phoen) [inline src/Analyzer/Pass/Statement/YodaCondition.php]: As long as they are not silently ignored, I'm okay with it :)

**Pull request merged**